crates/pixi/tests/integration_rust/pypi_tests.rs

```diff
@@ -873,3 +873,125 @@ async fn test_uv_index_correctly_parsed() {
             .contains(&simple.index_path().display().to_string())
     );
 }
+
+/// Tests that prerelease-mode = "allow" allows pre-release versions to be resolved.
+/// Without this setting, the resolver would skip pre-releases unless explicitly requested.
+#[tokio::test]
+async fn test_prerelease_mode_allow() {
+    setup_tracing();
+
+    // Build a local simple index with both a stable and prerelease version
+    let simple = PyPIDatabase::new()
+        .with(PyPIPackage::new("testpkg", "1.0.0"))
+        .with(PyPIPackage::new("testpkg", "2.0.0a1")) // Pre-release version
+        .into_simple_index()
+        .expect("failed to create local simple index");
+
+    let platform = Platform::current();
+
+    let mut package_db = PackageDatabase::default();
+    package_db.add_package(
+        Package::build("python", "3.12.0")
+            .with_subdir(platform)
+            .finish(),
+    );
+    let channel = package_db.into_channel().await.unwrap();
+    let channel_url = channel.url();
+
+    // With prerelease-mode = "allow", the resolver should pick the pre-release 2.0.0a1
```

Comment:
Do you think it makes sense to have a first test where we don't set any prerelease modes, and expect to get a simple 1.0.0 version?